core/instance.go

```diff
@@ -537,8 +537,11 @@ func (i *instance) createRunInstancesInput(instanceType string, price float64) *
 	if i.asg.LaunchTemplate != nil {
 		retval.LaunchTemplate = &ec2.LaunchTemplateSpecification{
 			LaunchTemplateId: i.asg.LaunchTemplate.LaunchTemplateId,
+			Version:          i.asg.LaunchTemplate.Version,
 		}
-	} else if i.IamInstanceProfile != nil {
```

Comment:
I think this was a bug, I doubt Launch Templates and the IAM Instance Profile should be mutually exclusive.